x-pack/plugins/cases/public/components/all_cases/selector_modal/index.tsx

```diff
@@ -4,91 +4,8 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import { AllCasesSelectorModal, AllCasesSelectorModalProps } from './all_cases_selector_modal';
 
-import React, { useState, useCallback } from 'react';
-import { EuiModal, EuiModalBody, EuiModalHeader, EuiModalHeaderTitle } from '@elastic/eui';
-import styled from 'styled-components';
-import {
-  Case,
-  CaseStatusWithAllStatus,
-  CommentRequestAlertType,
-  SubCase,
-} from '../../../../common';
-import { CasesNavigation } from '../../links';
-import * as i18n from '../../../common/translations';
-import { AllCasesGeneric } from '../all_cases_generic';
-import { Owner } from '../../../types';
-import { OwnerProvider } from '../../owner_context';
-
-export interface AllCasesSelectorModalProps extends Owner {
-  alertData?: Omit<CommentRequestAlertType, 'type'>;
-  createCaseNavigation: CasesNavigation;
-  hiddenStatuses?: CaseStatusWithAllStatus[];
-  onRowClick: (theCase?: Case | SubCase) => void;
-  updateCase?: (newCase: Case) => void;
-  userCanCrud: boolean;
-  onClose?: () => void;
-}
-
-const Modal = styled(EuiModal)`
-  ${({ theme }) => `
-    width: ${theme.eui.euiBreakpoints.l};
-    max-width: ${theme.eui.euiBreakpoints.l};
-  `}
-`;
-
-const AllCasesSelectorModalComponent: React.FC<AllCasesSelectorModalProps> = ({
-  alertData,
-  createCaseNavigation,
-  hiddenStatuses,
-  onRowClick,
-  updateCase,
-  userCanCrud,
-  onClose,
-}) => {
-  const [isModalOpen, setIsModalOpen] = useState<boolean>(true);
-  const closeModal = useCallback(() => {
-    if (onClose) {
-      onClose();
-    }
-    setIsModalOpen(false);
-  }, [onClose]);
-  const onClick = useCallback(
-    (theCase?: Case | SubCase) => {
-      closeModal();
-      onRowClick(theCase);
-    },
-    [closeModal, onRowClick]
-  );
-  return isModalOpen ? (
-    <Modal onClose={closeModal} data-test-subj="all-cases-modal">
-      <EuiModalHeader>
-        <EuiModalHeaderTitle>{i18n.SELECT_CASE_TITLE}</EuiModalHeaderTitle>
-      </EuiModalHeader>
-      <EuiModalBody>
-        <AllCasesGeneric
-          alertData={alertData}
-          createCaseNavigation={createCaseNavigation}
-          hiddenStatuses={hiddenStatuses}
-          isSelectorView={true}
-          onRowClick={onClick}
-          userCanCrud={userCanCrud}
-          updateCase={updateCase}
-        />
-      </EuiModalBody>
-    </Modal>
-  ) : null;
-};
-
-export const AllCasesSelectorModal: React.FC<AllCasesSelectorModalProps> = React.memo((props) => {
-  return (
-    <OwnerProvider owner={props.owner}>
-      <AllCasesSelectorModalComponent {...props} />
-    </OwnerProvider>
-  );
-});
-
-AllCasesSelectorModal.displayName = 'AllCasesSelectorModal';
-
+export { AllCasesSelectorModal, AllCasesSelectorModalProps };
 // eslint-disable-next-line import/no-default-export
 export { AllCasesSelectorModal as default };
```

Comment:
because it is lazy loaded in `x-pack/plugins/cases/public/methods/get_all_cases_selector_modal.tsx` it needs to be in the default export, the named `AllCasesSelectorModal` export is used by the test, but it could be removed.